cmd/image-builder-maintenance/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/ilyakaznacheev/cleanenv"
+
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	logrus.SetReportCaller(true)
+
+	var conf Config
+
+	err := cleanenv.ReadEnv(&conf)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	if conf.DryRun {
+		logrus.Info("Dry run, no state will be changed")
+	}
+
+	if !conf.EnableDBMaintenance {
+		logrus.Info("🦀🦀🦀 DB maintenance not enabled, skipping  🦀🦀🦀")
+		return
+	}
+	dbURL := fmt.Sprintf("postgres://%s:%s@%s:%s/%s?sslmode=%s",
+		conf.PGUser,
+		conf.PGPassword,
+		conf.PGHost,
+		conf.PGPort,
+		conf.PGDatabase,
+		conf.PGSSLMode,
+	)
+	err = DBCleanup(dbURL, conf.DryRun, conf.DBEmailRetentionMonths)
+	if err != nil {
+		logrus.Fatalf("Error during DBCleanup: %v", err)
+	}
+	logrus.Info("🦀🦀🦀 dbqueue cleanup done 🦀🦀🦀")
+}
```

Comment:
Would love a context to be created with a deadline and also SIGTERM handler goroutine sending cancel immediately in case the pod needs to be terminated for any reason. Then the SQL driver will properly shutdown the transaction.

Additionally, a correlation id could be generated, inserted into the context as value, [extracted in a simple hook](https://github.com/osbuild/osbuild-composer/blob/b5c71cd7e2ab31b5a3c2b8beeb31c3acf230301c/internal/common/context_hook.go) and as long as all logging statements start with `WithContext(ctx)` all logs will be also correlated and can be sent to kibana for archival.

Reply:
looking forward to a "correlation id" or trace_id … let's align on this separately. The context here should be prepared for it. :-)